src/rust/cryptography-x509/src/extensions.rs

```diff
@@ -18,15 +18,15 @@ pub type RawExtensions<'a> = common::Asn1ReadableOrWritable<
 ///
 /// In particular, an `Extensions` cannot be constructed from a `RawExtensions`
 /// that contains duplicated extensions (by OID).
-pub struct Extensions<'a>(Option<RawExtensions<'a>>);
+pub struct Extensions<'a, 'b>(Option<&'b RawExtensions<'a>>);
```

Comment:
This is adding an extra indirection. The `clones` are basically free, really just moving a few pointers (they are notably _not_ memory copies anywhere). If the motivation for this is just to remove `clone`, we should remove unless there's a benchmark showing value.

Reply:
Yeah, I have no numbers showing value here. I did this thinking that `RawExtensions` was a `Vec` under the hood and so avoiding a clone on it would be nice, but now I see that it's a `SequenceOf` so very little is gained by saving a clone here :slightly_smiling_face: